### ISSUE TITLE
%run_viz Jupyter line magic inconsistency

### ISSUE THREAD

**Issue by ravi-kumar-pilla** [kedro-org/kedro-viz#1823]
## Description

Tried executing %run_viz line magic on Jupyter notebook to test `--inlcude-hooks` option, while the command gets passed appropriately, Open Viz link appears even before considering the flag. This shows, we are not terminating the previous session.

This inconsistency was also detected during https://github.com/kedro-org/kedro-viz/pull/1733

![Image](https://github.com/kedro-org/kedro-viz/assets/87735534/25dd9774-07cc-4e5c-a132-e0a1507d0e87)

The image above is from a debug session using print statements placed at -
1. run_viz function at `kedro_viz/launchers/jupyter.py` line 120
2. checked if viz is up for the second run by placing a print statement at `kedro_viz/launchers/jupyter.py` line 133
3. placed a print statement in load_data function at `kedro_viz/integrations/kedro/data_loader.py` line 99

## Context

I am not an expert in using Jupyter notebook but when I execute `%run_viz` line magic with differing options, each execution box should take in the options before restarting the viz server. Currently, we check if the viz is up and we show the Open Kedro Viz link which has a stale session. 

## Steps to Reproduce

1. Setup Kedro Viz and have a Kedro Project
2. Execute `kedro jupyter notebook`
3. Create a notebook and execute `%run_viz`
4. It takes some time and gives you a link to open kedro viz
5. Try executing `%run_viz --ignore-hooks` or any other option and debug it using print statements
6. The link to open kedro viz appears instantly, without restarting the server with new arguments

## Actual Result

![Image](https://github.com/kedro-org/kedro-viz/assets/87735534/25dd9774-07cc-4e5c-a132-e0a1507d0e87)

## Your Environment
Include as many relevant details as possible about the environment you experienced the bug in:

* Web browser system and version: Chrome/Safari
* Operating system and version: macOS
* NodeJS version used (if relevant):
* Kedro version used (if relevant):
* Python version used (if relevant):

## Checklist
- [x] Include labels so that we can categorise your issue

**Comment by astrojuanlu** [comment]
If we succeed in #1993, maybe `%run_viz` is not so important?

**Comment by ravi-kumar-pilla** [comment]
> If we succeed in [#1993](https://github.com/kedro-org/kedro-viz/issues/1993), maybe `%run_viz` is not so important?

Yes, it might be in future. As of now #1993 focusses on simple pipeline objects rather than full blown kedro project. Since we are injecting json data into the html, we need to consider -

1. The html document size
2. Performance issues considering the <script> tag which holds the json can block html rendering 
3. XSS - Embedding large json directly inside html is not recommended (serving via an API endpoint is ideal)

Thank you

**Comment by astrojuanlu** [comment]
Let's check if this is still the case after https://github.com/kedro-org/kedro-viz/issues/2052 was addressed in 10.1.0
